server/php-runner.cpp

```diff
@@ -516,6 +513,7 @@ namespace kphp_runtime_signal_handlers {
 
 static void sigalrm_handler(int signum) {
   kwrite_str(2, "in sigalrm_handler\n");
+  vk::singleton<JsonLogger>::get().write_log_with_backtrace("Maximum execution time exceeded", E_ERROR);
   if (check_signal_critical_section(signum, "SIGALRM")) {
     PhpScript::tl_flag = true;
     // if script is not actually running, don't bother (sigalrm handler is called
```

Comment:
It needs to check critical section, running script state and `is_json_log_on_timeout_enabled` as it was before:
https://github.com/VKCOM/kphp/commit/e8729d8f60b3421160f50023f15c937878eb7719#diff-5e957cbea1d21c855029b0b0b8c112a7f07c01042584720bfc1659170b6bbe79L440